concordium-contracts-common/src/types.rs

```diff
@@ -1204,12 +1214,46 @@ impl From<Vec<u8>> for OwnedParameter {
     fn from(param: Vec<u8>) -> Self { Self(param) }
 }
 
+/// Display the entire parameter in hex.
+impl fmt::Display for OwnedParameter {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        for b in &self.0 {
+            f.write_fmt(format_args!("{:02x}", b))?
+        }
+        Ok(())
+    }
+}
+
 impl OwnedParameter {
     pub fn as_parameter(&self) -> Parameter { Parameter(self.0.as_ref()) }
 
     /// Construct an `OwnedParameter` by serializing the input using its
     /// `Serial` instance.
     pub fn new<D: Serial>(data: &D) -> Self { Self(to_bytes(data)) }
+
+    /// Construct a parameter from a vector of bytes without checking that it
+    /// fits the size limit. The caller is assumed to ensure this via
+    /// external means.
+    #[inline]
+    pub fn new_unchecked(bytes: Vec<u8>) -> Self { Self(bytes) }
+
+    /// Attempt to convert a byte array to a [`Self`].
+    ///
+    /// This will fail if the byte array is too large.
+    /// If it is known upfront that the array is less than
+    /// [`MAX_PARAMETER_LEN`][constants::MAX_PARAMETER_LEN] then it is better to
+    /// use [`Self::new_unchecked`].
+    pub fn try_from_bytes(bytes: Vec<u8>) -> Result<Self, ExceedsParameterSize> {
+        let actual = bytes.len();
+        if actual <= constants::MAX_PARAMETER_LEN {
+            Ok(Self::new_unchecked(bytes))
+        } else {
+            Err(ExceedsParameterSize {
+                actual,
+                max: constants::MAX_PARAMETER_LEN,
+            })
+        }
+    }
```

Comment:
I believe that the agreement was to
- implement `TryFrom` for OwnedParameter equivalent to the one we have for "Parameter" already.
- Remove the From<Vec<u8>> implementation for OwnedParameter (this is a breaking change in concordium-std, but that was agreement since it is unlikely to affect anybody.

Reply:
Great, that also seems like the best option.
I had just forgotten parts of the agreement ;)